renderdoc/driver/gl/gl_debug.cpp

```diff
@@ -379,17 +379,30 @@ void GLReplay::InitDebugData()
 
   DebugData.fixedcolFSProg = CreateShaderProgram(empty, fs);
 
-  GenerateGLSLShader(vs, shaderType, "", GetEmbeddedResource(glsl_mesh_vert), glslBaseVer);
-  GenerateGLSLShader(fs, shaderType, "", GetEmbeddedResource(glsl_mesh_frag), glslBaseVer);
-  GenerateGLSLShader(gs, shaderType, "", GetEmbeddedResource(glsl_mesh_geom), glslBaseVer);
+  if(HasExt[ARB_geometry_shader4])
+  {
+    GenerateGLSLShader(vs, shaderType, "", GetEmbeddedResource(glsl_mesh_vert), glslBaseVer);
+    GenerateGLSLShader(fs, shaderType, "", GetEmbeddedResource(glsl_mesh_frag), glslBaseVer);
+    GenerateGLSLShader(gs, shaderType, "", GetEmbeddedResource(glsl_mesh_geom), glslBaseVer);
 
-  DebugData.meshProg = CreateShaderProgram(vs, fs);
-  DebugData.meshgsProg = CreateShaderProgram(vs, fs, gs);
+    DebugData.meshProg = CreateShaderProgram(vs, fs);
+    DebugData.meshgsProg = CreateShaderProgram(vs, fs, gs);
 
-  GenerateGLSLShader(fs, shaderType, "", GetEmbeddedResource(glsl_trisize_frag), glslBaseVer);
-  GenerateGLSLShader(gs, shaderType, "", GetEmbeddedResource(glsl_trisize_geom), glslBaseVer);
+    GenerateGLSLShader(fs, shaderType, "", GetEmbeddedResource(glsl_trisize_frag), glslBaseVer);
+    GenerateGLSLShader(gs, shaderType, "", GetEmbeddedResource(glsl_trisize_geom), glslBaseVer);
 
-  DebugData.trisizeProg = CreateShaderProgram(vs, fs, gs);
+    DebugData.trisizeProg = CreateShaderProgram(vs, fs, gs);
+  }
+  else
+  {
+    GenerateGLSLShader(vs, shaderType, "", GetEmbeddedResource(glsl_mesh_vert), glslBaseVer);
+    GenerateGLSLShader(fs, shaderType, "", GetEmbeddedResource(glsl_mesh_frag), glslBaseVer);
+    GenerateGLSLShader(gs, shaderType, "", GetEmbeddedResource(glsl_mesh_geom), glslBaseVer);
+
+    DebugData.meshProg = CreateShaderProgram(vs, fs);
+    DebugData.meshgsProg = DebugData.meshProg;
+    DebugData.trisizeProg = DebugData.meshProg;
+  }
```

Comment:
It's not correct to set `meshgsProg` and `trisizeProg` to use the same program as `meshProg` since they do different things and expect different uniforms.

For other cases where an extension might be needed for some programs, I instead set them to 0 and add code in the places where they are used to either fall-back or not render depending on whether the feature is supported.